service/src/java/org/apache/hive/service/cli/thrift/ThriftHttpServlet.java

```diff
@@ -540,6 +556,35 @@ public String run() throws HttpAuthenticationException {
       }
     }
 
+    private void enforceLdapFilters(String shortName) throws HttpAuthenticationException {
+      boolean enableGroupCheck = hiveConf.getBoolVar(
+          HiveConf.ConfVars.HIVE_SERVER2_LDAP_ENABLE_GROUP_CHECK_AFTER_KERBEROS);
+      if (!enableGroupCheck) {
+        LOG.debug("No LDAP group check is enabled; skipping.");
+        return;
+      }
+      // Check if this is a proxy user - we don't apply LDAP filters to proxy users
+      String proxyUser = SessionManager.getProxyUserName();
```

Comment:
Here we are building the connection, I guess the `proxyUser` is always null. In any case do we see this value as a non-empty string?